### PR TITLE
ssd-resnet34 on COCO map 0.23

### PR DESCRIPTION
**Pull request by florescl**
https://wandb.ai/mosaic-ml/ssd-8gpu/runs/1xkziya0/overview?workspace=user-laura

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -21,7 +21,7 @@ style:
 # this only checks for style & pyright, makes no code changes
 lint:
 	$(PYTHON) -m isort -c --diff $(dirs)
-	$(PYTHON) -m yapf -dr $(dirs)
+	$(PYTHON) -m yapf -ri $(dirs)
```
Comment (hanlint) [inline Makefile:24]:
```suggestion
	$(PYTHON) -m yapf -dr $(dirs)
```

reverting this change -- `make lint` should not be in-place.

File: composer/datasets/coco.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+from composer.core.types import Batch, DataSpec, List
+from composer.datasets.dataloader import DataloaderHparams
+from composer.datasets.hparams import DatasetHparams
+from composer.models.ssd.utils import SSDTransformer, dboxes300_coco
+from composer.utils import dist
+
+
+@dataclass
+class COCODatasetHparams(DatasetHparams):
+    """Defines an instance of.
```
Comment (hanlint) [inline composer/datasets/coco.py:24]:
```suggestion
    """Defines an instance of COCO Dataset.
```

File: composer/models/ssd/hparams.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
+train_dataset:
+  coco:
+    is_train: true
+    datadir: /localdisk/coco
```
Comment (hanlint) [inline composer/models/ssd/hparams.yaml:4]:
```suggestion
    datadir: /datasets/coco
```

File: composer/models/ssd/hparams.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    download: false
+    shuffle: false
+    drop_last: false
+val_dataset:
+  coco:
+    is_train: false
+    datadir: /localdisk/coco
```
Comment (hanlint) [inline composer/models/ssd/hparams.yaml:11]:
```suggestion
    datadir: /datasets/coco
```

File: composer/models/ssd/hparams.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    lr: 0.08
+    weight_decay: 5e-4
+    momentum: 0.9
+model:
+  ssd:
+    input_size: 300
+    overlap_threshold: 0.5
+    nms_max_detections: 200
+    initializers: {}
+    data: /localdisk/coco
```
Comment (hanlint) [inline composer/models/ssd/hparams.yaml:26]:
```suggestion
    data: /datasets/coco
```

**Review by hanlint — APPROVED** [review]

File: setup.py [hunk 6]
```diff
@@ -54,8 +54,10 @@ def package_files(directory: str):
     "yahp>=0.0.14",
     "requests>=2.26.0",
     "numpy==1.21.5",
+    "pycocotools@git+https://github.com/cocodataset/cocoapi#egg=pycocotools&subdirectory=PythonAPI",
```
Comment (ravi-mosaicml) [inline setup.py:57]: This should not be a core dependency...can you make it a conditional install?

File: setup.py [hunk 7]
```diff
@@ -92,6 +94,7 @@ def package_files(directory: str):
     'pylint>=2.12.2',
     'docformatter>=1.4',
     'sphinx_panels==0.6.0',
+    'pycocotools@git+https://github.com/cocodataset/cocoapi#egg=pycocotools&subdirectory=PythonAPI',
```
Comment (ravi-mosaicml) [inline setup.py:97]: This should not need to be repeated.

**Pull request merged**